01_financial-well-being/income_metrics_county.qmd

```diff
@@ -0,0 +1,735 @@
+-   [Evaluate Results](#evaluate-results)
+-   [Export](#export)
+
+## Housekeeping
+
+Read in packages for use. If you do not have an IPUMS API key you will need to make one via the [IPUMS website](https://www.ipums.org/).
+
+```{r}
+library(tidyverse)
+library(Hmisc)
+library(ipumsr)
+library(reactable)
+library(srvyr)
+
+options(scipen = 999)
+
+#set_ipums_api_key("Your KEY", save = TRUE)
+
+source(here::here("functions", "API", "extract_ipums.R"))
+```
+
+
+## Read Data
+
+Read in the ACS extracts using the extract_ipums function. Make sure to change the survey list to reflect what years you want to include in the data. Currently we are including three years of 1-year data (2018a, 2021a and 2022a) and two years of 5-year data (2018c and 2021c). 5-year 2022 data will be added pending its release.
+```{r}
+acs <- extract_ipums(
+  extract_name = "umf_data_18_22",
+  extract_description = "Microdata pull for Mobility Metric Predictors. American Community Survey, years 2018, 2019 and 2022 (1 and 5-year).",
+  survey = list("us2018a", "us2021a", "us2022a", "us2018c", "us2021c")
+) 
+```
+
+Look at the distribution of samples in the data. The 5-year data sets should comprise a larger share of the total sample.
+```{r}
+acs %>% 
+  count(sample) %>% 
+  ggplot(mapping = aes(x = sample, y = n)) +
+  geom_col() +
+  theme_minimal() +
+  ggtitle("Samples, All Records") +
+ geom_text(mapping = aes(label = n), vjust = -1) +    
+  scale_y_continuous(expand = expansion(mult = c(0, 0.1))) +
+  labs(x = "Samples",
+       y = NULL)
+```
+
+Read in the PUMA to county crosswalk file. This file is created by the program generate_puma_county_crosswalks.rmd in the geographic-crosswalks folder. 
+```{r}
+puma_county_crosswalk <- read_csv(here::here("geographic-crosswalks", "data", "crosswalk_puma_to_county.csv"))
+```
+
+Crate a version of the crosswalk with just the county-level values of data quality variables for each crosswalk period.
+```{r}
+county_puma <- puma_county_crosswalk %>%
+  group_by(crosswalk_period, statefip, county) %>% 
+  summarise(geographic_allocation_quality = mean(geographic_allocation_quality)) %>% 
+  ungroup()
+```
+
+## Clean Data
+
+Missing data is reported in a variety of ways by IPUMS - listed below:
+
+HHINCOME is missing it is reported as "9999999".
+Vacancy is reported as "0". 
+
+There is a large share of records missing HHINCOME. However, note there are no vacancy results by default of the structure of census data read in.
+
+```{r}
+acs %>% 
+  ggplot(aes(x = hhincome)) +
+  geom_density(color = "blue", fill = alpha("blue", 0.3)) +
+  theme_minimal() +
+  ggtitle("Household Income Variable Results") +
+  ylab("Density")
+
+acs %>% 
+  ggplot(aes(x = vacancy)) +
+  geom_density(color = "blue", fill = alpha("blue", 0.3)) +
+  theme_minimal() +
+  ggtitle("Vacancy Variable Results") +
+  ylab("Density")
+```
+
+Turn the missing/"not in universe" values for the hhincome variable into NAs.
+
+```{r}
+
+acs_clean <- acs %>%
```

Comment:
In chunk 18, I get the following error otherwise:
Error: cannot allocate vector of size 653.0 Mb
Error during wrapup: cannot allocate vector of size 326.5 Mb
Error: no more error handlers available (recursive errors?); invoking 'abort' restart